lib/_http_server.js

```diff
@@ -447,20 +448,25 @@ function connectionListener(socket) {
       // if the user never called req.read(), and didn't pipe() or
       // .resume() or .on('data'), then we call req._dump() so that the
       // bytes will be pulled off the wire.
-      if (!req._consuming && !req._readableState.resumeScheduled)
+      if (!req._consuming && !req._readableState.resumeScheduled) {
         req._dump();
+      }
 
       res.detachSocket(socket);
+      res._last = outgoing.length === 0;
 
       if (res._last) {
-        socket.destroySoon();
+        // When connection sent last res server will wait for server.keepAliveTimeout
+        socket.removeAllListeners('timeout');
+        socket.setTimeout(self.keepAliveTimeout || 5000, function onKeepAliveTimeout() {
```

Comment:
That listener will fire 3 "timeout" events for req, res and server, but in this case we need no those events, socket will be closed in normal way, not timed out, not hanging.